coding-projects/ios/TaskTracker/TaskTracker/List Screen/ListView.swift

```diff
@@ -8,11 +8,39 @@
 import SwiftUI
 
 struct ListView: View {
+    @ObservedObject var viewModel = DetailsViewModel()
+    
+    // Load activities when the view appears
+    init() {
+        viewModel.loadActivities()
+    }
+    
     var body: some View {
-        Text(/*@START_MENU_TOKEN@*/"Hello, World!"/*@END_MENU_TOKEN@*/)
+        NavigationView {
+            List {
+                // Iterate over grouped activities
+                ForEach(viewModel.groupedActivities.keys.sorted(), id: \.self) { date in
+                    Section(header: Text(date).bold()) {
+                        // Display each activity in this group
+                        ForEach(viewModel.groupedActivities[date] ?? [], id: \.id) { activity in
```

Comment:
```suggestion
                        ForEach(viewModel.groupedActivities[date] ?? [], id: \.id) { activity in
```

Is there a way to prevent the use of a nil coalescing operator here. Just seems that we shouldn't be checking for an empty array in here but should've been done before. 

I don't know if it's actually possible, but just asking.